### PR TITLE
Separates the menu for multi-select of motions

### PR DESCRIPTION
**Pull request by GabrielInTheWorld**
- Adds a new component for the list.
- Adds the component to amendment-list and motion-list.

### CODE REVIEW

File: client/src/app/shared/components/motion-menu-list/motion-menu-list.component.html [hunk 1]
```diff
@@ -0,0 +1,81 @@
+<button
```
Comment (FinnStutzenstein) [inline client/src/app/shared/components/motion-menu-list/motion-menu-list.component.html]: Note: Please move all motion related components away from the shared folder.

File: client/src/app/site/motions/modules/amendment-list/amendment-list.component.html [hunk 2]
```diff
@@ -136,5 +136,15 @@ <h2>
             <mat-icon>clear</mat-icon>
             <span translate>Deselect all</span>
         </button>
+
+        <div *osPerms="'motions.can_manage'; or: 'motions.can_manage_metadata'">
+            <mat-divider></mat-divider>
+            <os-motion-menu-list [selectedMotions]="selectedRows" (action)="multiselectWrapper($event)">
+                <button mat-menu-item (click)="openExportDialog()">
```
Comment (tsiegleauq) [inline client/src/app/site/motions/modules/amendment-list/amendment-list.component.html]: please remove and move into `os-motion-menu-list`.
Just like in the normal motion list, the list of amendments should also be able to export selected motions

File: client/src/app/site/motions/modules/motion-list/components/motion-list/motion-list.component.html [hunk 3]
```diff
@@ -327,104 +327,33 @@
...
+                    <span translate>Move in call list</span>
+                </button>
+                <button
+                    *ngIf="perms.isAllowed('manage')"
+                    mat-menu-item
+                    [disabled]="!selectedRows.length"
+                    (click)="openExportDialog()"
+                >
+                    <mat-icon>archive</mat-icon>
+                    <span translate>Export selected motions</span>
```
Comment (tsiegleauq) [inline client/src/app/site/motions/modules/motion-list/components/motion-list/motion-list.component.html]: as above - Export selected motions for both the motion list and the amendment list

File: client/src/app/site/motions/modules/shared-motion/motion-menu-list/motion-menu-list.component.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+import { TagRepositoryService } from 'app/core/repositories/tags/tag-repository.service';
+import { ConfigService } from 'app/core/ui-services/config.service';
+import { ViewCategory } from 'app/site/motions/models/view-category';
+import { ViewMotion } from 'app/site/motions/models/view-motion';
+import { ViewMotionBlock } from 'app/site/motions/models/view-motion-block';
+import { MotionMultiselectService } from 'app/site/motions/services/motion-multiselect.service';
+import { ViewTag } from 'app/site/tags/models/view-tag';
+
+@Component({
+    selector: 'os-motion-menu-list',
```
Comment (tsiegleauq) [inline client/src/app/site/motions/modules/shared-motion/motion-menu-list/motion-menu-list.component.ts]: should probably be
`os-motion-multiselect-actions`

File: client/src/app/site/motions/modules/shared-motion/motion-menu-list/motion-menu-list.component.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    templateUrl: './motion-menu-list.component.html',
+    styleUrls: ['./motion-menu-list.component.scss']
+})
+export class MotionMenuListComponent implements OnInit, OnDestroy {
```
Comment (tsiegleauq) [inline client/src/app/site/motions/modules/shared-motion/motion-menu-list/motion-menu-list.component.ts]: extending `BaseViewComponent` should be fine here

File: client/src/app/site/motions/modules/shared-motion/motion-menu-list/motion-menu-list.component.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    /**
+     * The list of all motion-blocks.
+     */
+    public motionBlocks: ViewMotionBlock[] = [];
+
+    /**
+     * A list to store subscriptions, until the component is destroyed.
+     * In `ngOnDestroy` all subscriptions will be unsubscribed.
+     */
+    private subscriptions: Subscription[] = [];
```
Comment (tsiegleauq) [inline client/src/app/site/motions/modules/shared-motion/motion-menu-list/motion-menu-list.component.ts]: can be removed if extending BaseViewComponent

File: client/src/app/site/motions/modules/shared-motion/motion-menu-list/motion-menu-list.component.ts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+            })
+        );
+    }
+
+    /**
+     * OnDestroy-method.
+     *
+     * Unsubscribes all subscriptions and reset the array to default.
+     */
+    public ngOnDestroy(): void {
```
Comment (tsiegleauq) [inline client/src/app/site/motions/modules/shared-motion/motion-menu-list/motion-menu-list.component.ts]: can be removed if extending BaseViewComponent

**Pull request merged**